### ISSUE TITLE
Dropdown A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#6326]
Tracking all of the issues brought up with the 09/10 accessibility pass on Drop Down control
Details on individual bugs can be found on  [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B6fd5c0ed-4367-456d-ab08-8e7ace591f1f%7D&action=default&uid=%7B6FD5C0ED-4367-456D-AB08-8E7ACE591F1F%7D&ListItemId=1340&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- [ ] ~~Bug 1 : [Accessibility][Keyboard navigation]: Tool tip of list items in drop down are not accessible with keyboard.~~

>   WON'T FIX - screen readers will read title. And for non-screen-readers, this goes against users' expectations

- [x] Bug 2 : [Accessibility][Usable]: Narrator does not convey the 'Expand/Collapsed' state of 'Basic uncontrolled drop down'.
- [x] Bug 3 : [Accessibility][Programmatic access]: 'Error message' is not associated with the 'Error message example drop down'

> Tracked in #10600

- [x] Bug 4 : [Accessibility][Keyboard navigation]: Keyboard Focus indicator is not visible on 'Error message example drop down'.

> Tracked in #10636

- [ ] ~~Bug 5 : [Accessibility][Keyboard navigation]: When user navigates on 'Basic uncontrolled example drop down' first option gets selected by itself.~~

> This is actually adhering to accessibility specs. See #5473 and https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_kbd_interaction.

- [x] Bug 6 : [Accessibility][Usable]: Narrator scan mode focus is not visible on 'Basic uncontrolled example' drop down when it is collapsed. 
- [x] Bug 7 : [Accessibility][Keyboard navigation]: Arrow key navigation is not proper while navigating inside the 'Basic uncontrolled example' Drop down.
- [x] Bug 8 : [Accessibility][Usable]: Narrator speaks Improper information on 'Customized Dropdown'.  
- [x] Bug 9 : [Accessibility][Supporting the platform] : Selected Drop down list items are not visible in Black/white high contrast theme setting. 
- [x] Bug  10 : [Accessibility][Programmatic access]: Narrator does not convey role of the drop downs. 

> Tracked in [#6892]

- [ ] ~~Bug 11 : [Accessibility][Programmatic access]: Position of check boxes is not defined inside 'Multi-select controlled/uncontrolled example dropdowns'.~~

> Unable to repro (not clear what "position of check boxes is not defined" mean)

- [ ] ~~Bug 12 : [Accessibility][Usable] : In scan mode while navigating list items, 'Controlled example dropdown' does not get closed if user presses tab key.~~

> Most of the time when I press tab key, the focus jumps OFF the webpage into UI elements of the browser chrome. Focus on the webpage still remains on the Dropdown. In instances where focus did not jump to the browser chrome and moved on to another element in the webpage, the Dropdown does close.

- [x] Bug 13 : [Accessibility][Screen reader]: In scan mode user is not able to select the 'English' list item under Multi-Select uncontrolled example drop down' in downward navigation.
- [ ] ~~Bug 14 : [Accessibility][Programmatic access]: When user selects the 'Set focus button' focus jumps to the first list item of 'Basic uncontrolled example' where narrator does not convey the label pf the dropdown.~~

> Does not repro at https://fabricweb.z5.web.core.windows.net/oufr/7.43.0/#/examples/dropdown

- [x] Bug 15 : [Accessibility][Forms and validation]: Group labels inside dropdowns are not associated with its corresponding control.
- [x] Bug 16 : [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation. 
- [x] Bug 17 : [Accessibility][Screen reader]: Narrator focus moves twice on the first element of context menu in scan mode. 
- [x] Bug 18 : [Accessibility][Screen reader]: Narrator focus moves onto the separator line in scan mode. 
- [ ] ~~Bug 19 : [Accessibility][Screen reader]: In Scan mode user is not able to close the 'Contextual menu' using 'Esc key'.~~

> Does not repro - pressing esc does close the "Contextual menu" (I assume this reers to the Dropdown popup - unsure what Contextual Menu means otherwise in the context of Dropdown)

- [x] Bug 20 : [Accessibility][Screen reader]: Narrator focus moves onto hidden elements  in scan mode in backward navigation. 

@micahgodbolt 

- [ ] ~~Bug 21 : [Accessibility][Dropdown]: Elements in the Drop down screen are not adopting to 400% scaling~~

> Does not repro. (400% zoom below) (p.s. in screenshot below, GitHub has resized the image but it is 400% zoom)
![image](https://user-images.githubusercontent.com/1003246/65717570-8b46e800-e056-11e9-95bf-d1165f1ca19b.png)
> On smaller resolutions, it looks like this:
> ![image](https://user-images.githubusercontent.com/1003246/65730978-be01d800-e078-11e9-8d69-867fc9e31089.png)

- [ ] ~~Bug 22 : [Accessibility][Dropdown]: Focus indicator is not visible on "Set Focus" button in High Contrast white and black modes~~

> Does not repro. As you can see in the screenshot below, there is a focus rectangle on the Set focus button.  
![image](https://user-images.githubusercontent.com/1003246/65718231-ffce5680-e057-11e9-901b-53ba25920f00.png)

**Comment by micahgodbolt** [comment]
@TestAccessibilityGIT thanks for the update!

**Comment by cliffkoh** [comment]
Bug 10 is #5752

**Comment by micahgodbolt** [comment]
#7300 is bug 2

**Comment by micahgodbolt** [comment]
@JasonGore can you take a peak at bug 2? It is specifically affecting a partner.

**Comment by micahgodbolt** [comment]
bug 2 is a narrator bug. you'll see the same behavior on this example https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html

**Comment by jamesedjac** [comment]
Can we get an update on bug 19?

**Comment by micahgodbolt** [comment]
Nothing right now. Scan mode is a really difficult mode to work in when it comes to interaction. If you have any ideas, or working examples of closing menus with scan mode, we'd love to see them.

**Comment by jamesedjac** [comment]
That makes sense. I imagine the issue is that dropdown is not a formal role in ARIA so narrator is not honoring the custom ESC command. I will reach out to the Narrator team on this

**Comment by Kumar2608** [comment]
Bug 6 : [Accessibility][Usable]: Narrator scan mode focus is not visible on 'Basic uncontrolled example' drop down when it is collapsed.
In scan mode, focus is visible on "Basic uncontrolled example" drop down.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

Bug 8 : [Accessibility][Usable]: Narrator speaks Improper information on 'Customized Dropdown'.
Narrator is providing the appropriate information on "Customized Drop down'.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

Bug 9 : [Accessibility][Supporting the platform] : Selected Drop down list items are not visible in Black/white high contrast theme setting.
Selected Drop down list items are clearly visible in High Contrast White and Black modes.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

Bug 13 : [Accessibility][Screen reader]: In scan mode user is not able to select the 'English' list item under Multi-Select uncontrolled example drop down' in downward navigation.
In scan mode, user is able to access all the list elements under Multi-Select uncontrolled example dropdown" in downward direction.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

  Bug 15 : [Accessibility][Forms and validation]: Group labels inside dropdowns are not associated with its corresponding control.
Group labels inside dropdowns are associated with its corresponding control.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

  Bug 16 : [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation.
Screen is scrolling down along with keyboard focus in back navigation.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

  Bug 17 : [Accessibility][Screen reader]: Narrator focus moves twice on the first element of context menu in scan mode.
Narrator focus moves only once on the first element of context menu in scan mode.
Issue is fixed, Verified the issue in the below build:
https://fabricweb.z5.web.core.windows.net/oufr/7.31.2/#/examples/dropdown

**Comment by Kumar2608** [comment]
**New WCAG 2.1 Bugs and Bugs Found During Regression**

Bug 21 : [Accessibility][Dropdown]: Drop down screen is not adopting to 400% scaling
Bug 22 : [Accessibility][Dropdown]: Focus indicator is not visible  on "Set Focus" button in High Contrast white and black modes

**Comment by cliffkoh** [comment]
@TestAccessibilityGIT  @Kumar2608 I've gone through to triage most of them (but still have a couple more to go). Could you please add more info to bugs  19, 21, 11 as I am unable to repro?

@jamesedjac Actually I cannot repro bug 19. Pressing esc key while in scan mode does close the Dropdown for me for the Dropdown at https://fabricweb.z5.web.core.windows.net/oufr/7.39.1/#/examples/dropdown

**Comment by cliffkoh** [comment]
Hi @Kumar2608  @TestAccessibilityGIT, all items above have been triaged or fixed. I am going to close this issue as more than 7 days have passed without a response. If there is still a concern, please kindly open a new issue.

**Comment by 999-Dev** [comment]
Bug 1 : [Accessibility][Keyboard navigation]: Tool tip of list items in drop down are not accessible with keyboard.
We observe that tooltip information and name of item is same and as discussed this is very low impact.
OS: Version 1903(OS Build 19005.1)
Browser: Edge
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/dropdown
Hence closing the bug.

**Comment by 999-Dev** [comment]
Bug 3 : [Accessibility][Programmatic access]: 'Error message' is not associated with the 'Error message example drop down'
We observe that UI has been changed now. 
OS: Version 1903(OS Build 19005.1)
Browser: Edge
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/dropdown
Hence closing the bug.

**Comment by 999-Dev** [comment]
Bug 2 : [Accessibility][Usable]: Narrator does not convey the 'Expand/Collapsed' state of 'Basic uncontrolled drop down'.
We observe that screen reader announces the collapse state of the dropdown control.
 
**Build Version:**
OS: Version 1903(OS Build 19005.1)
Browser: Edge
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/dropdown

Hence closing the bug.
